libraries/networking/src/udt/PacketList.cpp

```diff
@@ -53,6 +80,23 @@ std::unique_ptr<Packet> PacketList::createPacketWithExtendedHeader() {
     return packet;
 }
 
+QByteArray PacketList::getAllData() {
+    size_t sizeBytes = 0;
+
+    for (const auto& packet : _packets) {
+        sizeBytes += packet->size();
+    }
+
+    QByteArray data;
+    data.reserve(sizeBytes);
+
+    for (auto& packet : _packets) {
+        data.append(packet->getPayload(), packet->getPayloadSize());
+    }
+
+    return data;
+}
```

Comment:
I think it's a little confusing that `getDataSize` includes the headers but `getAllData` does not.

Maybe `getMessage` is a better name?